test/suites/clustering.sh

```diff
@@ -3228,66 +3228,122 @@ test_clustering_remove_members() {
   LXD_DIR="${LXD_TWO_DIR}" lxc info --target node1 | grep -q "server_name: node1"
+  LXD_DIR="${LXD_THREE_DIR}" lxc info --target node1 | grep -q "server_name: node1"
+  LXD_DIR="${LXD_FOUR_DIR}" lxc info --target node1 | grep -q "server_name: node1"
+  LXD_DIR="${LXD_FIVE_DIR}" lxc info --target node1 | grep -q "server_name: node1"
+  LXD_DIR="${LXD_SIX_DIR}" lxc info --target node1 | grep -q "server_name: node1"
 
-  # Remove the leader, via the stand-by node
-  LXD_DIR="${LXD_TWO_DIR}" lxc cluster rm node1
+  # stop node 6
+  shutdown_lxd "${LXD_SIX_DIR}"
+
+  # Remove node2 node3 node4 node5
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster rm node2
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster rm node3
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster rm node4
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster rm node5
 
   # Ensure the remaining node is working
-  ! LXD_DIR="${LXD_TWO_DIR}" lxc cluster list | grep -q "node1" || false
-  LXD_DIR="${LXD_TWO_DIR}" lxc cluster list | grep -q "node2"
+  ! LXD_DIR="${LXD_ONE_DIR}" lxc cluster list | grep -q "node2" || false
+  ! LXD_DIR="${LXD_ONE_DIR}" lxc cluster list | grep -q "node3" || false
+  ! LXD_DIR="${LXD_ONE_DIR}" lxc cluster list | grep -q "node4" || false
+  ! LXD_DIR="${LXD_ONE_DIR}" lxc cluster list | grep -q "node5" || false
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster list | grep -q "node1"
 
-  # Previous leader should no longer be clustered
-  ! LXD_DIR="${LXD_ONE_DIR}" lxc cluster list || false
+  # Start node 6
+  LXD_NETNS="${ns6}" respawn_lxd "${LXD_SIX_DIR}" true
 
-  # Spawn a third node, joining the cluster with node2
-  setup_clustering_netns 3
-  LXD_THREE_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
-  chmod +x "${LXD_THREE_DIR}"
-  ns3="${prefix}3"
-  spawn_lxd_and_join_cluster "${ns3}" "${bridge}" "${cert}" 3 2 "${LXD_THREE_DIR}"
+  # make sure node 6 is spare
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster list | grep -q "node6"
+  ! LXD_DIR="${LXD_ONE_DIR}" lxc cluster show node6 | grep -qE "\- database-standy$|\- database-leader$|\- database$" || false
 
-  # Ensure successful communication
-  LXD_DIR="${LXD_TWO_DIR}" lxc info --target node3 | grep -q "server_name: node3"
-  LXD_DIR="${LXD_THREE_DIR}" lxc info --target node2 | grep -q "server_name: node2"
+  LXD_DIR="${LXD_ONE_DIR}" lxc cluster list
+  # waite for new leader ready
+  sleep 10s
 
-  # Remove the third node, via itself
-  LXD_DIR="${LXD_THREE_DIR}" lxc cluster rm node3
+  # Remove the leader, via the stand-by node
+  LXD_DIR="${LXD_SIX_DIR}" lxc cluster rm node1
 
-  # Ensure only node2 is left in the cluster
-  LXD_DIR="${LXD_TWO_DIR}" lxc cluster ls | grep -q node2
-  ! LXD_DIR="${LXD_TWO_DIR}" lxc cluster ls -f csv | grep -qv node2 || false
+  # Ensure the remaining node is working
+  ! LXD_DIR="${LXD_SIX_DIR}" lxc cluster list | grep -q "node1" || false
+  LXD_DIR="${LXD_SIX_DIR}" lxc cluster list | grep -q "node6"
 
-  # Ensure clustering is disabled (no output) on node1 and node3
```

Comment:
We seem to have lost some scenarios that were being tested before is this due to the change in logic?

Reply:
This test is to construct a scenario where a cluster only have a spare node and leader node，then remove the leader，check other node become to leader and cluster is still work by join a new node，so i think my test include the scenarios what old test caase。